packages/accounts/src/msca/plugins/multisig/actions/proposeUserOperation.ts

```diff
@@ -52,11 +58,15 @@ export async function proposeUserOperation<
     throw new SmartAccountWithSignerRequiredError();
   }
 
-  const builtUo = await client.buildUserOperation({
-    account,
-    uo,
-    overrides,
-  });
+  const builtUo = await client
+    .buildUserOperation({
+      account,
+      uo,
+      overrides,
+    })
+    .then((x) => multisigSignatureMiddleware(x, { account, client, overrides }))
```

Comment:
@dphilipson @denniswon want to revisit the argument against having the signer middleware run with all functions that call the middleware? It seems weird to have signing middleware not run as part of the stack and leads to some gnarly maintenance costs. It's too easy to forget to call the `client.middleware.signer` where it's needed.

I'd like to add it back. If the signer allows signing without prompts, then not having it part of the build UO flow doesn't add much safety. So there's not a compelling security argument against it.

Reply:
no, that is incorrect. It should not run per middleware stack.

checkSponsorshipEligiblity, DropAndReplace, simulateUserOperation should not necessarily have the user operation request signed. 

cc: @dphilipson

Reply:
see the thread for context as to why it should be in the middleware stack as an opt-in

Reply:
Caught up with that thread https://alchemyinsights.slack.com/archives/C0669LA6JQH/p1713888009511439


However, I still don't understand why signing needs to be in the middleware stack. sign user operation was added as a middleware not because we want all user op building/estimating/simulation etc to sign, but as a mechanism to have custom signing at the control of the user.

It seems like just for propose user op for multi sig, where we want it to be part of the middleware stack? so this is the only "opt-in" case right? Why can't we utilize custom Middleware ? isn't that custom middleware's purpose?

Reply:
> sign user operation was added as a middleware not because we want all user op building/estimating/simulation etc to sign

when I added it that actually was the intent and the original assumption that `proposeUserOperation` made. I've update the approach in this PR for how propose etc works to leverage `signUserOperation`. But if we want to use `signUserOperation` as a method then maybe middleware isn't the best place for that custom logic to live. we can probs actually move it out of middleware entirely and allow it to be an overridable action... lemme take a look at doing that actually cuz I hate it in the middleware object if it's not run as part of the middleware stack

Reply:
> lemme take a look at doing that actually cuz I hate it in the middleware object if it's not run as part of the middleware stack

never mind, this isn't super easy to do actually because of the order in which actions get added it's not possible for the base actions to use overrides. gonna leave it as middleware even though it has no business being middleware anymore